dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyPath.java

```diff
@@ -37,7 +37,9 @@
 
   @VisibleForTesting
   public void add(Dependency dependency) {
-    path.add(dependency);
+    // null optional is translated to false
+    boolean isOptional = dependency.getOptional() != null && dependency.getOptional();
```

Comment:
Is there some reason this needs to be set explicitly here? Why can we just pass through a null?

Reply:
This is explicitly set as false because `(dependency.getOptional() ? ", optional" : "")` threw NullPointerException in [formatDependency ](https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/1229/files#diff-45d74402f4f801feb0a96b570930f212R89)